### PR TITLE
DOCSUP-7574: Add slowdown_count and show actual errors_count in the system.clusters table

### PR DESCRIPTION
**Pull request by sevirov**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category:

- Documentation for #21480

### CODE REVIEW

File: docs/ru/operations/system-tables/clusters.md [hunk 1]
```diff
@@ -4,12 +4,70 @@
...
+-   `shard_num` ([UInt32](../../sql-reference/data-types/int-uint.md)) — номер шарда в кластере, начиная с 1.
+-   `shard_weight` ([UInt32](../../sql-reference/data-types/int-uint.md)) — относительный вес шарда при записи данных.
+-   `replica_num` ([UInt32](../../sql-reference/data-types/int-uint.md)) — номер реплики в шарде, начиная с 1.
+-   `host_name` ([String](../../sql-reference/data-types/string.md)) — хост, указанный в конфигурации.
+-   `host_address` ([String](../../sql-reference/data-types/string.md)) — TIP-адрес хоста, полученный из DNS.
+-   `port` ([UInt16](../../sql-reference/data-types/int-uint.md)) — порт, на который обращаться для соединения с сервером.
+-   `is_local` ([UInt8](../../sql-reference/data-types/int-uint.md)) — флаг, показывающий является ли хост локальным.
+-   `user` ([String](../../sql-reference/data-types/string.md)) — имя пользователя, которого использовать для соединения с сервером.
+-   `default_database` ([String](../../sql-reference/data-types/string.md)) — имя базы данных по умолчанию.
+-   `errors_count` ([UInt32](../../sql-reference/data-types/int-uint.md)) — количество неудачных попыток у хоста получить доступ к реплике.
```
Comment (adevyatova) [inline docs/ru/operations/system-tables/clusters.md:17]:
```suggestion
-   `errors_count` ([UInt32](../../sql-reference/data-types/int-uint.md)) — количество неудачных попыток хоста получить доступ к реплике.
```

File: docs/ru/operations/system-tables/clusters.md [hunk 2]
```diff
@@ -4,12 +4,70 @@
...
+-   `slowdowns_count` ([UInt32](../../sql-reference/data-types/int-uint.md)) — количество раз, когда была заменена реплика при установке соединения при хеджированных запросах из-за долгого отсутствия ответа от реплики.
+-   `estimated_recovery_time` ([UInt32](../../sql-reference/data-types/int-uint.md)) — количество секунд до того момента, когда количество ошибок будет обнулено и реплика будет доступной.
```
Comment (adevyatova) [inline docs/ru/operations/system-tables/clusters.md:19]:
```suggestion
-   `estimated_recovery_time` ([UInt32](../../sql-reference/data-types/int-uint.md)) — количество секунд до момента, когда количество ошибок будет обнулено и реплика станет доступной.
```

File: docs/ru/operations/system-tables/clusters.md [hunk 3]
```diff
@@ -4,12 +4,70 @@
...
 
+Обратите внимание, что `errors_count` обновляется один раз при запросе к кластеру, а `estimated_recovery_time` пересчитывается по требованию. Таким образом, может возникнуть ситуация, при которой `errors_count` не равен нулю, а `estimated_recovery_time` равен нулю. Поэтому следующий запрос обнулит `errors_count` и попытается использовать реплику так, как будто у нее нет ошибок.
```
Comment (adevyatova) [inline docs/ru/operations/system-tables/clusters.md:21]: Выдели это как примечание.

File: docs/ru/operations/system-tables/clusters.md [hunk 4]
```diff
@@ -4,12 +4,70 @@
...
+-   `replica_num` ([UInt32](../../sql-reference/data-types/int-uint.md)) — номер реплики в шарде, начиная с 1.
+-   `host_name` ([String](../../sql-reference/data-types/string.md)) — хост, указанный в конфигурации.
+-   `host_address` ([String](../../sql-reference/data-types/string.md)) — TIP-адрес хоста, полученный из DNS.
+-   `port` ([UInt16](../../sql-reference/data-types/int-uint.md)) — порт, на который обращаться для соединения с сервером.
+-   `is_local` ([UInt8](../../sql-reference/data-types/int-uint.md)) — флаг, показывающий является ли хост локальным.
+-   `user` ([String](../../sql-reference/data-types/string.md)) — имя пользователя, которого использовать для соединения с сервером.
+-   `default_database` ([String](../../sql-reference/data-types/string.md)) — имя базы данных по умолчанию.
+-   `errors_count` ([UInt32](../../sql-reference/data-types/int-uint.md)) — количество неудачных попыток у хоста получить доступ к реплике.
+-   `slowdowns_count` ([UInt32](../../sql-reference/data-types/int-uint.md)) — количество раз, когда была заменена реплика при установке соединения при хеджированных запросах из-за долгого отсутствия ответа от реплики.
```
Comment (adevyatova) [inline docs/ru/operations/system-tables/clusters.md:18]:
```suggestion
-   `slowdowns_count` ([UInt32](../../sql-reference/data-types/int-uint.md)) — количество замен реплики из-за долгого отсутствия ответа от нее при установке соединения при хеджированных запросах.
```

File: docs/ru/operations/system-tables/clusters.md [hunk 5]
```diff
@@ -4,12 +4,70 @@
...
--   `user` (String) — имя пользователя, которого использовать для соединения с сервером.
+-   `cluster` ([String](../../sql-reference/data-types/string.md)) — имя кластера.
+-   `shard_num` ([UInt32](../../sql-reference/data-types/int-uint.md)) — номер шарда в кластере, начиная с 1.
+-   `shard_weight` ([UInt32](../../sql-reference/data-types/int-uint.md)) — относительный вес шарда при записи данных.
+-   `replica_num` ([UInt32](../../sql-reference/data-types/int-uint.md)) — номер реплики в шарде, начиная с 1.
+-   `host_name` ([String](../../sql-reference/data-types/string.md)) — хост, указанный в конфигурации.
+-   `host_address` ([String](../../sql-reference/data-types/string.md)) — TIP-адрес хоста, полученный из DNS.
+-   `port` ([UInt16](../../sql-reference/data-types/int-uint.md)) — порт, на который обращаться для соединения с сервером.
+-   `is_local` ([UInt8](../../sql-reference/data-types/int-uint.md)) — флаг, показывающий является ли хост локальным.
+-   `user` ([String](../../sql-reference/data-types/string.md)) — имя пользователя, которого использовать для соединения с сервером.
```
Comment (adevyatova) [inline docs/ru/operations/system-tables/clusters.md:15]:
```suggestion
-   `user` ([String](../../sql-reference/data-types/string.md)) — имя пользователя для соединения с сервером.
```

File: docs/ru/operations/system-tables/clusters.md [hunk 6]
```diff
@@ -4,12 +4,70 @@
...
--   `host_address` (String) — TIP-адрес хоста, полученный из DNS.
--   `port` (UInt16) — порт, на который обращаться для соединения с сервером.
--   `user` (String) — имя пользователя, которого использовать для соединения с сервером.
+-   `cluster` ([String](../../sql-reference/data-types/string.md)) — имя кластера.
+-   `shard_num` ([UInt32](../../sql-reference/data-types/int-uint.md)) — номер шарда в кластере, начиная с 1.
+-   `shard_weight` ([UInt32](../../sql-reference/data-types/int-uint.md)) — относительный вес шарда при записи данных.
+-   `replica_num` ([UInt32](../../sql-reference/data-types/int-uint.md)) — номер реплики в шарде, начиная с 1.
+-   `host_name` ([String](../../sql-reference/data-types/string.md)) — хост, указанный в конфигурации.
+-   `host_address` ([String](../../sql-reference/data-types/string.md)) — TIP-адрес хоста, полученный из DNS.
+-   `port` ([UInt16](../../sql-reference/data-types/int-uint.md)) — порт, на который обращаться для соединения с сервером.
```
Comment (adevyatova) [inline docs/ru/operations/system-tables/clusters.md:13]:
```suggestion
-   `port` ([UInt16](../../sql-reference/data-types/int-uint.md)) — порт для соединения с сервером.
```

File: docs/en/operations/system-tables/clusters.md [hunk 7]
```diff
@@ -4,63 +4,70 @@ Contains information about clusters available in the config file and the servers
...
+-   `host_name` ([String](../../sql-reference/data-types/string.md)) — The host name, as specified in the config.
+-   `host_address` ([String](../../sql-reference/data-types/string.md)) — The host IP address obtained from DNS.
+-   `port` ([UInt16](../../sql-reference/data-types/int-uint.md)) — The port to use for connecting to the server.
+-   `is_local` ([UInt8](../../sql-reference/data-types/int-uint.md)) — Flag that indicates whether the host is local.
+-   `user` ([String](../../sql-reference/data-types/string.md)) — The name of the user for connecting to the server.
+-   `default_database` ([String](../../sql-reference/data-types/string.md)) — The default database name.
+-   `errors_count` ([UInt32](../../sql-reference/data-types/int-uint.md)) — The number of times this host failed to reach replica.
+-   `slowdowns_count` ([UInt32](../../sql-reference/data-types/int-uint.md)) — The number of slowdowns that led to changing replica when establishing a connection with hedged requests.
+-   `estimated_recovery_time` ([UInt32](../../sql-reference/data-types/int-uint.md)) — Seconds left until replica error count is zeroed and it is considered to be back to normal.
```
Comment (adevyatova) [inline docs/en/operations/system-tables/clusters.md:19]:
```suggestion
-   `estimated_recovery_time` ([UInt32](../../sql-reference/data-types/int-uint.md)) — Seconds remaining until the replica error count is zeroed and it is considered to be back to normal.
```

**Comment by sevirov** [comment]
@avogar Hi. Please, review this PR.

File: docs/en/operations/system-tables/clusters.md [hunk 8]
```diff
@@ -4,63 +4,71 @@ Contains information about clusters available in the config file and the servers
...
 
-Please note that `errors_count` is updated once per query to the cluster, but `estimated_recovery_time` is recalculated on-demand. So there could be a case of non-zero `errors_count` and zero `estimated_recovery_time`, that next query will zero `errors_count` and try to use replica as if it has no errors.
-
-**See also**
-
--   [Table engine Distributed](../../engines/table-engines/special/distributed.md)
--   [distributed_replica_error_cap setting](../../operations/settings/settings.md#settings-distributed_replica_error_cap)
--   [distributed_replica_error_half_life setting](../../operations/settings/settings.md#settings-distributed_replica_error_half_life)
+!!! note "Note"
+    Please note that `errors_count` is updated once per query to the cluster, but `estimated_recovery_time` is recalculated on-demand. So there could be a case of non-zero `errors_count` and zero `estimated_recovery_time`, that next query will zero `errors_count` and try to use replica as if it has no errors.
```
Comment (Avogar) [inline docs/en/operations/system-tables/clusters.md:22]: Now this `Note` is irrelevant since we update `errors_count` on-demand now.

**Review by Avogar — APPROVED** [review]

**Pull request merged**